src/cryptography/hazmat/bindings/openssl/binding.py

```diff
@@ -139,3 +140,19 @@ def _lock_cb(cls, mode, n, file, line):
                     mode, n, file, line
                 )
             )
+
+
+# https://github.com/pyca/cryptography/issues/2299 contains extensive
+# discussion about this workaround. In essence, we are acquiring a global lock
+# (including across all subinterpreters if we're in a mod_wsgi environment)
+# initializing some C level locks, then initializing the binding and adding the
+# OS random engine. This should prevent engine addition race conditions,
+# although it's still unclear what happens when a subinterpreter which
+# has registered the locks is destroyed. Do the callback functions still work
+# or are all the objects associated with that subinterpreter destroyed?
+try:
```

Comment:
@glyph can probably more clearly articulate this, but my understanding is that Python 3.4+ uses per module locks rather than the global import lock so while for Py < 3.4 we could just call `init_static_locks` and `_ensure_ffi_initialized` at module scope and be safe, 3.4+ would require us to explicitly acquire the global import lock.